frontend/src/metabase/visualizations/lib/apply_tooltips.js

```diff
@@ -6,160 +6,207 @@ import d3 from "d3";
 import { formatValue } from "metabase/lib/formatting";
 import type { ClickObject } from "metabase/meta/types/Visualization"
 
+import { isNormalized, isStacked } from "./renderer_utils";
 import { determineSeriesIndexFromElement } from "./tooltip";
 import { getFriendlyName } from "./utils";
 
-export function applyChartTooltips(chart, series, isStacked, isNormalized, isScalarSeries, onHoverChange, onVisualizationClick) {
-    let [{ data: { cols } }] = series;
-    chart.on("renderlet.tooltips", function(chart) {
-        chart.selectAll("title").remove();
+function getHoverDataForScatterChart(d, cols) {
+    if (d.key._origin) {
+        return d.key._origin.row.map((value, index) => {
+            const col = d.key._origin.cols[index];
+            return { key: getFriendlyName(col), value: value, col };
+        });
+    }
+    return d.key.map((value, index) => (
+        { key: getFriendlyName(cols[index]), value: value, col: cols[index] }
+    ));
+}
+
+function getHoverDataForLineAreaBarChart(d, i, thisSeries, cols, isNormalized) {
```

Comment:
`thisSeries` is an interesting param name. Mind elaborating a bit on why you chose it?

Reply:
So as you noticed below `series` is the param used through the charting lib to mean an array of one or more series. Which, I think we can agree, is really confusing. But it's not immediately clear to me what we should do about it because there's no plural version (`serieses`?)

So to make it clear we're dealing with a single series I have been playing around with things like `thisSeries` when we pass it. Something like `aSeries` could work too